desc/grid.py

```diff
@@ -422,12 +422,12 @@ def __init__(self, nodes, sort=False, jitable=False):
             r = jnp.where(r == 0, 1e-12, r)
             self._nodes = jnp.array([r, t, z]).T
             self._axis = np.array([], dtype=int)
-            self._unique_rho_idx = np.arange(self._nodes.shape[0])
-            self._unique_theta_idx = np.arange(self._nodes.shape[0])
-            self._unique_zeta_idx = np.arange(self._nodes.shape[0])
-            self._inverse_rho_idx = np.arange(self._nodes.shape[0])
-            self._inverse_theta_idx = np.arange(self._nodes.shape[0])
-            self._inverse_zeta_idx = np.arange(self._nodes.shape[0])
+            self._unique_rho_idx = jnp.argsort(self._nodes[:, 0])
```

Comment:
something I did not realize, when jitable is passed the unique rho indices are not actually computed?

Reply:
no, since the size of the array wouldn't be a compile time constant. This works assuming all the rho/theta/zeta values are unique but not if there are any on the same surface.